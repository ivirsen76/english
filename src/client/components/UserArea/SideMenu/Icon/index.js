import React from 'react'
import PropTypes from 'prop-types'
import style from './style.module.scss'

export default class Icon extends React.Component {
    static propTypes = {
        type: PropTypes.string,
    }

    render() {
        return <i className={`icon-${this.props.type} ${style.icon}`} />
    }
}
