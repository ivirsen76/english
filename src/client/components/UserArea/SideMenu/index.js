import React from 'react'
import PropTypes from 'prop-types'
import { Menu } from 'semantic-ui-react'
import MenuLink from 'components/MenuLink'

const SideMenu = ({ cardTotal, cardRememberTotal, cardWriteTotal }) => (
    <Menu vertical fluid size="large">
        <Menu.Item>
            <Menu.Header>Словарь</Menu.Header>

            <Menu.Menu>
                <MenuLink to="/user/cards">
                    Мои слова
                    <div className="ui label">{cardTotal}</div>
                </MenuLink>
            </Menu.Menu>
        </Menu.Item>

        <Menu.Item>
            <Menu.Header>Изучение</Menu.Header>

            <Menu.Menu>
                <MenuLink to="/user/remember">
                    Запоминание
                    <div className="ui label">{cardRememberTotal}</div>
                </MenuLink>
                <MenuLink to="/user/write">
                    Написание
                    <div className="ui label">{cardWriteTotal}</div>
                </MenuLink>
                <MenuLink to="/user/mp3">
                    Скачать MP3
                </MenuLink>
            </Menu.Menu>
        </Menu.Item>

        <Menu.Item>
            <Menu.Header>Справочно</Menu.Header>

            <Menu.Menu>
                <MenuLink to="/user/statistics">
                    Статистика
                </MenuLink>
            </Menu.Menu>
        </Menu.Item>

        <Menu.Item>
            <Menu.Header>Настройки</Menu.Header>

            <Menu.Menu>
                <MenuLink to="/user/tasks">
                    Задачи
                </MenuLink>
                <MenuLink to="/user/base">
                    База слов
                </MenuLink>
            </Menu.Menu>
        </Menu.Item>
    </Menu>
)

SideMenu.propTypes = {
    cardTotal: PropTypes.number,
    cardRememberTotal: PropTypes.number,
    cardWriteTotal: PropTypes.number,
}

export default SideMenu
