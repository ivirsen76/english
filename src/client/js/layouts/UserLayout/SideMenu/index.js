import React from 'react'
import PropTypes from 'prop-types'
import { Menu } from 'semantic-ui-react'
import MenuLink from 'client/js/components/MenuLink'
import { hasRole } from 'client/js/utils/auth.js'
import IconDownload from '@ieremeev/icons/download'
import IconBooks from '@ieremeev/icons/books'
import IconBubble from '@ieremeev/icons/bubble2'
import IconPencil from '@ieremeev/icons/pencil2'
import IconStats from '@ieremeev/icons/stats-dots'
import IconLibrary from '@ieremeev/icons/library'

const SideMenu = ({ cardTotal, cardRememberTotal, cardWriteTotal }) => (
    <Menu id="sideMenu" vertical fluid size="huge">
        <Menu.Item>
            <Menu.Header>Словарь</Menu.Header>

            <Menu.Menu>
                <MenuLink to="/user/cards">
                    <IconBooks spaceRight />
                    Мои слова
                    <div className="ui label" id="cardTotal">
                        {cardTotal}
                    </div>
                </MenuLink>
                <MenuLink to="/user/baseBrowser">
                    <IconDownload spaceRight />
                    Добавить из баз
                </MenuLink>
            </Menu.Menu>
        </Menu.Item>

        <Menu.Item>
            <Menu.Header>Обучение</Menu.Header>

            <Menu.Menu>
                <MenuLink id="rememberMenuItem" to="/user/remember">
                    <IconBubble spaceRight />
                    Запомнить
                    <div className="ui label">{cardRememberTotal}</div>
                </MenuLink>
                <MenuLink id="writeMenuItem" to="/user/write">
                    <IconPencil spaceRight />
                    Написать
                    <div className="ui label">{cardWriteTotal}</div>
                </MenuLink>
                {/* <MenuLink to="/user/mp3"> */}
                {/*     <Icon type="headphones" /> */}
                {/*     Скачать MP3 */}
                {/* </MenuLink> */}
            </Menu.Menu>
        </Menu.Item>

        {hasRole('admin') && (
            <Menu.Item>
                <Menu.Header>Другое</Menu.Header>

                <Menu.Menu>
                    <MenuLink to="/user/statistics">
                        <IconStats spaceRight />
                        Статистика
                    </MenuLink>
                    <MenuLink to="/user/base">
                        <IconLibrary spaceRight />
                        Базы
                    </MenuLink>
                </Menu.Menu>
            </Menu.Item>
        )}
    </Menu>
)

SideMenu.propTypes = {
    cardTotal: PropTypes.number,
    cardRememberTotal: PropTypes.number,
    cardWriteTotal: PropTypes.number,
}

export default SideMenu
