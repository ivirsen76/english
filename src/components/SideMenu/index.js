import React, { PropTypes } from 'react'
import { Menu } from 'semantic-ui-react'
import MenuLink from 'components/MenuLink'


const SideMenu = ({ cardTotal }) => (
    <Menu vertical fluid size="large">
        <Menu.Item>
            <Menu.Header>Словарь</Menu.Header>

            <Menu.Menu>
                <MenuLink to="/user/cards">
                    Мои слова
                    <div className="ui label">{cardTotal}</div>
                </MenuLink>
            </Menu.Menu>
        </Menu.Item>

        <Menu.Item>
            <Menu.Header>Изучение</Menu.Header>

            <Menu.Menu>
                <MenuLink to="/user/remember">
                    Запоминание
                    <div className="ui label">12</div>
                </MenuLink>
                <MenuLink to="/user/write">
                    Написание
                    <div className="ui label">7</div>
                </MenuLink>
                <MenuLink to="/user/mp3">
                    Скачать MP3
                </MenuLink>
            </Menu.Menu>
        </Menu.Item>

        <Menu.Item>
            <Menu.Header>Справочно</Menu.Header>

            <Menu.Menu>
                <MenuLink to="/user/statistics">
                    Статистика
                </MenuLink>
            </Menu.Menu>
        </Menu.Item>

        <Menu.Item>
            <Menu.Header>Настройки</Menu.Header>

            <Menu.Menu>
                <MenuLink to="/user/tasks">
                    Задачи
                </MenuLink>
                <MenuLink to="/user/base">
                    База слов
                </MenuLink>
            </Menu.Menu>
        </Menu.Item>
    </Menu>
)

SideMenu.propTypes = {
    cardTotal: PropTypes.number,
}

export default SideMenu
