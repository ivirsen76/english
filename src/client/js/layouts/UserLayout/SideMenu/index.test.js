import React from 'react'
import { shallow } from 'enzyme'
import SideMenu from './index'

describe('<SideMenu />', () => {
    it('Should render menu', () => {
        const wrapper = shallow(
            <SideMenu cardTotal={3} cardRememberTotal={12} cardWriteTotal={7} />
        )
        expect(wrapper).toMatchSnapshot()
    })
})
