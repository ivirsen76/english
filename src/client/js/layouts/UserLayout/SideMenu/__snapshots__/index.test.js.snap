// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideMenu /> Should render menu 1`] = `
<Menu
  fluid={true}
  size="huge"
  vertical={true}
>
  <MenuItem>
    <MenuHeader>
      Vocabulary
    </MenuHeader>
    <MenuMenu>
      <withProps(NavLink)
        to="/user/cards"
      >
        <Icon
          type="books"
        />
        My cards
        <div
          className="ui label"
        >
          3
        </div>
      </withProps(NavLink)>
    </MenuMenu>
  </MenuItem>
  <MenuItem>
    <MenuHeader>
      Learning
    </MenuHeader>
    <MenuMenu>
      <withProps(NavLink)
        to="/user/remember"
      >
        <Icon
          type="bubble2"
        />
        Remember
        <div
          className="ui label"
        >
          12
        </div>
      </withProps(NavLink)>
      <withProps(NavLink)
        to="/user/write"
      >
        <Icon
          type="pencil2"
        />
        Write
        <div
          className="ui label"
        >
          7
        </div>
      </withProps(NavLink)>
      <withProps(NavLink)
        to="/user/mp3"
      >
        <Icon
          type="headphones"
        />
        Download MP3
      </withProps(NavLink)>
    </MenuMenu>
  </MenuItem>
  <MenuItem>
    <MenuHeader>
      Other
    </MenuHeader>
    <MenuMenu>
      <withProps(NavLink)
        to="/user/statistics"
      >
        <Icon
          type="stats-dots"
        />
        Statistics
      </withProps(NavLink)>
      <withProps(NavLink)
        to="/user/base"
      >
        <Icon
          type="library"
        />
        Bases
      </withProps(NavLink)>
    </MenuMenu>
  </MenuItem>
</Menu>
`;
